File: 1689.partitioning-into-minimum-number-of-deci-binary-numbers.ts
/*
 * @lc app=leetcode id=1689 lang=typescript
 *
 * [1689] Partitioning Into Minimum Number Of Deci-Binary Numbers
 */

// @lc code=start
function minPartitions(n: string): number {
  var result: string = "0";

  for (let num of n) {
    if (num > result) {
      result = num;
    }
  }
  return Number(result);
}
// @lc code=end

File: 1689.partitioning-into-minimum-number-of-deci-binary-numbers.cpp
/*
 * @lc app=leetcode id=1689 lang=cpp
 *
 * [1689] Partitioning Into Minimum Number Of Deci-Binary Numbers
 */

// @lc code=start
class Solution
{
public:
    int minPartitions(string n)
    {
        char result = '0';
        for (char digit : n)
        {
            result = max(result, digit);
        }
        return result - '0';
    }
};
// @lc code=end
